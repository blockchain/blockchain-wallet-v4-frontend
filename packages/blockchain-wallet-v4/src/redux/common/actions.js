import * as T from './actionTypes'

export const setBlockchainData = (data) =>
  ({ type: T.SET_BLOCKCHAIN_DATA, payload: data })
