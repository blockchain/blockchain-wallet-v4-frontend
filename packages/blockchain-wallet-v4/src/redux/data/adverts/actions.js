import * as T from './actionTypes'

export const setAdverts = (data) => ({ type: T.SET_ADVERTS, payload: data })
