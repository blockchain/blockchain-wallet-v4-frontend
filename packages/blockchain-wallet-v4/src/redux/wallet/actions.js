import * as T from './actionTypes'

// setters
export const setWrapper = payload => ({
  type: T.SET_WRAPPER,
  payload: payload
})
// ignored by the sync middleware and used by websocket middleware
export const refreshWrapper = payload => ({
  type: T.REFRESH_WRAPPER,
  payload: payload
})
export const setMainPassword = password => ({
  type: T.SET_MAIN_PASSWORD,
  payload: password
})
export const setPayloadChecksum = checksum => ({
  type: T.SET_PAYLOAD_CHECKSUM,
  payload: checksum
})
export const setLegacyAddressLabel = (address, label) => ({
  type: T.SET_LEGACY_ADDRESS_LABEL,
  payload: { address, label }
})
export const setAddressArchived = (address, archived) => ({
  type: T.SET_ADDRESS_ARCHIVED,
  payload: { address, archived }
})
export const setHdAddressLabel = (accountIdx, addressIdx, label) => ({
  type: T.SET_HD_ADDRESS_LABEL,
  payload: { accountIdx, addressIdx, label }
})
export const createLegacyAddress = address => ({
  type: T.CREATE_LEGACY_ADDRESS,
  payload: address
})
export const setAccountLabel = (accountIdx, label) => ({
  type: T.SET_ACCOUNT_LABEL,
  payload: { accountIdx, label }
})
export const setAccountArchived = (accountIdx, archived) => ({
  type: T.SET_ACCOUNT_ARCHIVED,
  payload: { accountIdx, archived }
})
export const setDefaultAccountIdx = index => ({
  type: T.SET_DEFAULT_ACCOUNT,
  payload: { index }
})
export const setTransactionNote = (txHash, txNote) => ({
  type: T.SET_TRANSACTION_NOTE,
  payload: { txHash, txNote }
})
export const setSyncPubKeys = syncPubKeys => ({
  type: T.SET_SYNC_PUB_KEYS,
  payload: { syncPubKeys }
})

// deletes
export const deleteWrapper = () => ({ type: T.DELETE_WRAPPER })
export const deleteLegacyAddress = address => ({
  type: T.DELETE_LEGACY_ADDRESS,
  payload: address
})
export const deleteHdAddressLabel = (accountIdx, addressIdx) => ({
  type: T.DELETE_HD_ADDRESS_LABEL,
  payload: { accountIdx, addressIdx }
})

// autologout time
export const setAutoLogout = time => ({
  type: T.SET_AUTOLOGOUT,
  payload: { time }
})

// mnemonic verified
export const verifyMnemonic = () => ({ type: T.VERIFY_MNEMONIC })
