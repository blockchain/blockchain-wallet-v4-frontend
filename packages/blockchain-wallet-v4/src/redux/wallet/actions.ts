import * as T from './actionTypes'

// setters
export const setWrapper = (payload) => ({
  payload,
  type: T.SET_WRAPPER
})
// ignored by the sync middleware and used by websocket middleware
export const refreshWrapper = (payload) => ({
  payload,
  type: T.REFRESH_WRAPPER
})
export const setMainPassword = (password) => ({
  payload: password,
  type: T.SET_MAIN_PASSWORD
})
export const setPayloadChecksum = (checksum) => ({
  payload: checksum,
  type: T.SET_PAYLOAD_CHECKSUM
})
export const setLegacyAddressLabel = (address, label) => ({
  payload: { address, label },
  type: T.SET_LEGACY_ADDRESS_LABEL
})
export const setAddressArchived = (address, archived) => ({
  payload: { address, archived },
  type: T.SET_ADDRESS_ARCHIVED
})
export const setHdAddressLabel = (accountIdx, addressIdx, derivationType, label) => ({
  payload: { accountIdx, addressIdx, derivationType, label },

  type: T.SET_HD_ADDRESS_LABEL
})
export const createLegacyAddress = (address) => ({
  payload: address,
  type: T.CREATE_LEGACY_ADDRESS
})
export const setAccountLabel = (accountIdx, label) => ({
  payload: { accountIdx, label },
  type: T.SET_ACCOUNT_LABEL
})
export const setAccountArchived = (accountIdx, archived) => ({
  payload: { accountIdx, archived },
  type: T.SET_ACCOUNT_ARCHIVED
})
export const setAccountDerivations = (accountIdx, derivations) => ({
  payload: { accountIdx, derivations },
  type: T.SET_ACCOUNT_DERIVATIONS
})
export const setDefaultAccountIdx = (index) => ({
  payload: { index },
  type: T.SET_DEFAULT_ACCOUNT
})
export const setTransactionNote = (txHash, txNote) => ({
  payload: { txHash, txNote },
  type: T.SET_TRANSACTION_NOTE
})
export const setSyncPubKeys = (syncPubKeys) => ({
  payload: { syncPubKeys },
  type: T.SET_SYNC_PUB_KEYS
})

// deletes
export const deleteWrapper = () => ({ type: T.DELETE_WRAPPER })
export const deleteLegacyAddress = (address) => ({
  payload: address,
  type: T.DELETE_LEGACY_ADDRESS
})
export const deleteHdAddressLabel = (accountIdx, addressIdx, derivationType) => ({
  payload: { accountIdx, addressIdx, derivationType },
  type: T.DELETE_HD_ADDRESS_LABEL
})

// autologout time
export const setAutoLogout = (time) => ({
  payload: { time },
  type: T.SET_AUTOLOGOUT
})

// mnemonic verified
export const verifyMnemonic = () => ({ type: T.VERIFY_MNEMONIC })

// check/update account labels
export const checkAndUpdateAccountLabels = () => ({
  type: T.CHECK_UPDATE_ACCT_LABELS
})

// mnemonic verified timestamp
export const updateMnemonicBackup = () => ({
  type: T.UPDATE_MNEMONIC_BACKUP
})

// trigger alert that mnemonic was viewed
export const triggerMnemonicViewedAlert = () => ({
  type: T.MNEMONIC_VIEWED_ALERT
})

export const triggerNonCustodialSendAlert = (currency: string, amount: any) => ({
  payload: { amount, currency },
  type: T.NON_CUSTODIAL_SEND_ALERT
})
