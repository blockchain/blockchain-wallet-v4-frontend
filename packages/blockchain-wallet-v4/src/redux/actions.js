import * as data from './data/actions'
import * as wallet from './wallet/actions'
import * as settings from './settings/actions'
import * as walletSync from './walletSync/actions'
import * as walletOptions from './walletOptions/actions'
import * as kvStore from './kvStore/actions'

export { data, kvStore, settings, wallet, walletSync, walletOptions }
