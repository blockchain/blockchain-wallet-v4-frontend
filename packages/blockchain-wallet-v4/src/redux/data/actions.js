import * as bch from './bch/actions'
import * as btc from './btc/actions'
import * as bsv from './bsv/actions'
import * as coinify from './coinify/actions'
import * as eth from './eth/actions'
import * as misc from './misc/actions'
import * as sfox from './sfox/actions'
import * as shapeShift from './shapeShift/actions'
import * as xlm from './xlm/actions'

// TODO: rename exports as coin codes
export {
  bch,
  btc as bitcoin,
  bsv,
  coinify,
  eth as ethereum,
  misc,
  sfox,
  shapeShift,
  xlm
}
