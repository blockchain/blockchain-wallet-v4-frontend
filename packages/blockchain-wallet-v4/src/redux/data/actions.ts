import * as algo from './algo/actions'
import * as bch from './bch/actions'
import * as btc from './btc/actions'
import * as dot from './dot/actions'
import * as eth from './eth/actions'
import * as fiat from './fiat/actions'
import * as misc from './misc/actions'
import * as stx from './stx/actions'
import * as xlm from './xlm/actions'

export { algo, bch, btc, dot, eth, fiat, misc, stx, xlm }
