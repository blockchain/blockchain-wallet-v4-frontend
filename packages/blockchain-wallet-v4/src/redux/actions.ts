import * as data from './data/actions'
import * as kvStore from './kvStore/actions'
import * as settings from './settings/actions'
import * as wallet from './wallet/actions'
import * as walletOptions from './walletOptions/actions'
import * as walletSync from './walletSync/actions'

export { data, kvStore, settings, wallet, walletOptions, walletSync }
