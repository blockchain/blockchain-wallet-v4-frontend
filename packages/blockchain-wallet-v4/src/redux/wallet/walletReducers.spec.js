import { compose } from 'ramda'
import { Wrapper, Wallet, AddressMap } from '../../types'
import walletReducer from './reducers.js'
import * as Actions from './actions.js'
const walletFixture = require('../../types/__mocks__/wallet.v3')

const wrap = wallet => ({
  sync_pubkeys: false,
  payload_checksum: 'payload_checksum',
  storage_token: 'storage_token',
  version: 3,
  language: 'en',
  wallet: wallet,
  war_checksum: 'war_checksum',
  password: 'password',
  pbkdf2_iterations: 5000
})

describe('reducers', () => {
  describe('wallet', () => {
    const wrapped = Wrapper.fromJS(wrap(walletFixture))

    it('should handle SET_WRAPPER', () => {
      let action = Actions.setWrapper(wrapped)
      let next = walletReducer(void 0, action)
      expect(next).toEqual(wrapped)
    })

    // it('should handle TOGGLE_SECOND_PASSWORD_SUCCESS ON', () => {
    //   let action = Actions.toggleSecondPasswordSuccess(wrapped)
    //   let next = walletReducer(void 0, action)
    //   expect(next).toEqual(wrapped)
    // })
    //
    // it('should handle CREATE_LEGACY_ADDRESS', () => {
    //   let action = Actions.createAddressSuccess(wrapped)
    //   let next = walletReducer(void 0, action)
    //   expect(next).toEqual(wrapped)
    // })

    it('should handle SET_LEGACY_ADDRESS_LABEL', () => {
      let label = 'my new label'
      let action = Actions.setLegacyAddressLabel(
        '19XmKRY66VnUn5irHAafyoTfiwFuGLUxKF',
        label
      )
      let next = walletReducer(wrapped, action)
      const sa = compose(
        AddressMap.selectAddress('19XmKRY66VnUn5irHAafyoTfiwFuGLUxKF'),
        Wallet.selectAddresses,
        Wrapper.selectWallet
      )
      expect(sa(next).label).toEqual(label)
    })

    it('should handle SET_HD_ADDRESS_LABEL', () => {
      let label = 'changed_label'
      let action = Actions.setHdAddressLabel(0, 0, label)
      let next = walletReducer(wrapped, action)
      let select = compose(
        Wallet.toJS,
        Wrapper.selectWallet
      )
      expect(
        select(next).hd_wallets[0].accounts[0].address_labels[0].label
      ).toEqual(label)
    })

    it('should handle DELETE_HD_ADDRESS_LABEL', () => {
      let action = Actions.deleteHdAddressLabel(0, 0)
      let next = walletReducer(wrapped, action)
      let select = compose(
        Wallet.toJS,
        Wrapper.selectWallet
      )
      expect(
        select(next).hd_wallets[0].accounts[0].address_labels.length
      ).toEqual(0)
    })

    // TODO :: wallet creation need review
    // it('should handle WALLET_NEW_SET', () => {
    //   let { wallet, mnemonic } = walletNewFixture
    //   let action = Actions.setNewWallet(wallet.guid, wallet.sharedKey, mnemonic)
    //   let next = walletImmutable(void 0, action)
    //   expect(Wallet.toJS(next)).to.deep.equal(wallet)
    // })
  })
})
