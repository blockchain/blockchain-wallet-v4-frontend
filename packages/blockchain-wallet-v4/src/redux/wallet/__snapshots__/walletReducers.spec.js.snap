// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers wallet should handle MERGE_WRAPPER 1`] = `
Immutable.Map {
  "sync_pubkeys": false,
  "payload_checksum": "payload_checksum",
  "storage_token": "storage_token",
  "version": 3,
  "language": "en",
  "wallet": Immutable.Map {
    "addresses": Immutable.Map {
      "19XmKRY66VnUn5irHAafyoTfiwFuGLUxKF": Immutable.Map {
        "addr": "19XmKRY66VnUn5irHAafyoTfiwFuGLUxKF",
        "priv": "38W3vsxt246Lhawvz81y2HQVnsJrBUX87jSTUUnixPsE",
        "tag": 0,
        "label": "",
        "created_time": 1492721419269,
        "created_device_name": "javascript_web",
        "created_device_version": "3.0",
      },
      "14mQxLtEagsS8gYsdWJbzthFFuPDqDgtxQ": Immutable.Map {
        "addr": "14mQxLtEagsS8gYsdWJbzthFFuPDqDgtxQ",
        "priv": "BpD2ZuJjZ8PJPpDX6ZmsKFsXHkL7XV3dt385zghMfF6C",
        "tag": 0,
        "label": "labeled_imported",
        "created_time": 1492721432222,
        "created_device_name": "javascript_web",
        "created_device_version": "3.0",
      },
      "1JD73aGSdeqKUjJ4ntP4eCyUiuZ3ogJE1u": Immutable.Map {
        "addr": "1JD73aGSdeqKUjJ4ntP4eCyUiuZ3ogJE1u",
        "priv": null,
        "tag": 0,
        "label": "",
        "created_time": 1492721461228,
        "created_device_name": "javascript_web",
        "created_device_version": "3.0",
      },
    },
    "tx_notes": Immutable.Map {},
    "guid": "new guid",
    "metadataHDNode": "xprv9tygGQP8be7uzNm5Czuy41juTK9pUKnWyZtDxgbmSEcCYa9VdvvtSknEyiKitqqm2TMv14NjXPQ68XLwSdH6Scc5GwXoZ31yRZZysxhVGU7",
    "tx_names": Immutable.List [],
    "double_encryption": false,
    "address_book": Immutable.Map {},
    "hd_wallets": Immutable.List [
      Immutable.Map {
        "seedHex": "6a4d9524d413fdf69ca1b5664d1d6db0",
        "passphrase": "",
        "mnemonic_verified": false,
        "default_account_idx": 0,
        "accounts": Immutable.List [
          Immutable.Map {
            "label": "My Bitcoin Wallet",
            "archived": false,
            "xpriv": "xprv9yL1ousLjQQzGNBAYykaT8J3U626NV6zbLYkRv8rvUDpY4f1RnrvAXQneGXC9UNuNvGXX4j6oHBK5KiV2hKevRxY5ntis212oxjEL11ysuG",
            "xpub": "xpub6CKNDRQEZmyHUrFdf1HapGEn27ramwpqxZUMEJYUUokoQrz9yLBAiKjGVWDuiCT39udj1r3whqQN89Tar5KrojH8oqSy7ytzJKW8gwmhwD3",
            "address_labels": Immutable.Map {
              "0": Immutable.Map {
                "index": 0,
                "label": "labeled_address",
              },
            },
            "cache": Immutable.Map {
              "receiveAccount": "xpub6F41z8MqNcJMvKQgAd5QE2QYo32cocYigWp1D8726ykMmaMqvtqLkvuL1NqGuUJvU3aWyJaV2J4V6sD7Pv59J3tYGZdYRSx8gU7EG8ZuPSY",
              "changeAccount": "xpub6F41z8MqNcJMwmeUExdCv7UXvYBEgQB29SWq9jyxuZ7WefmSTWcwXB6NRAJkGCkB3L1Eu4ttzWnPVKZ6REissrQ4i6p8gTi9j5YwDLxmZ8p",
            },
            "index": 0,
          },
        ],
      },
    ],
    "sharedKey": "8a260b2b-5257-4357-ac56-7a7efca323ea",
    "options": Immutable.Map {
      "pbkdf2_iterations": 5000,
      "fee_per_kb": 10000,
      "html5_notifications": false,
      "logout_time": 600000,
    },
  },
  "war_checksum": "war_checksum",
  "password": "password",
  "pbkdf2_iterations": 5000,
}
`;
